### PR TITLE
feat: support generating a color combination from a color palette

### PR DESCRIPTION
**Pull request by arjunlalb**
## Description
Adding support for generating a color combination from a string id.
This is to be used when a constant color has to be produced for an object based on its ID without having to define an explicit association.

Defining a string id to number hashing logic also to support the above requirement.

### CODE REVIEW

**Review by aaron-steinfeld — APPROVED** [review]

**Review by aaron-steinfeld — COMMENTED** [review]
Could be a worth a couple tests 😉

**Comment by arjunlalb** [comment]
> Could be a worth a couple tests 😉

Sure. Adding them now. I was waiting for the approach to be reviewed before I added tests.

**Review by aaron-steinfeld — APPROVED** [review]

File: projects/common/src/color/color-palette.test.ts [hunk 1]
```diff
@@ -31,4 +31,26 @@ describe('Color palette', () => {
...
+      {
+        background: 'rgb(20, 3, 0)',
+        foreground: '#FFFFFF'
+      }
+    ]);
+  });
+
+  test('should generate color for a string as expected from a limited set', () => {
+    const palette = new ColorPalette(['#fffbeb', '#140300', '#789ab7']);
+    expect(palette.getColorCombinationForId(2, 'test')).toEqual({
```
Comment (aaron-steinfeld) [inline projects/common/src/color/color-palette.test.ts:51]: Looking at how this is used, we could probably make the count optional and the second param. so just `getColorCombinationForString('test')`

Comment (arjunlalb) [reply projects/common/src/color/color-palette.test.ts:51]: makes sense. updated @aaron-steinfeld

**Review by aaron-steinfeld — DISMISSED** [review]

**Comment by aaron-steinfeld** [comment]
> @aaron-steinfeld accepted your suggestion. reapproval please.

damn, I failed prettier :(

**Pull request merged**